content/posts/why-meson/index.md

```diff
@@ -0,0 +1,174 @@
+flags for compiling a single file.
+
+# What we like about the Meson Build System
+
+- It is declarative, which means you don't have to remember or care about how
+to actually compile a shared library or a static library on Linux, Windows,
+BSD, etc. or how to link an executable with some other libraries or make sure
+include paths are right. As an example:
+```
+library('io',
+  ['file1.cpp', 'file2.cpp'],
+  dependencies: [util_dep],
+  install: true,
+  soversion: rz_asm_lib.version()
+)
+```
+You don't need to know *how* meson is going to build your library, but it is
+going to do it by compiling two source files (e.g. `file1.cpp` and
+`file2.cpp`), name the library `io` (e.g. on Linux the library would be
+called `libio.so`, but the full name and the extensions might be different on
+Windows) and give it the proper API version, make sure the dependency
+specified by `util_dep`, whatever it is, is used to compile this library, by
+adding the proper include paths and link directives.
+- It is *fast*. ACR/Make, due to its complexity as implemented in
+radare2/Rizin, does not correctly check which files are modified and need to
+be recompiled. Indeed, if you do `./configure;make` and then you re-run
+`make`, many files and operations will be re-done, even if nothing has
+changed. `meson` on the other hand keeps track of what is changed and it only
+re-builds those files that actually changed and other build targets that
+depend on them. This is extremely important for developers, as while
+developing a feature or a bug they may need to compile Rizin multiple times.
+- `meson` can run everywhere `python3` can. This includes a very wide range
+of platforms nowadays. It automatically provides a very powerful scripting
+language, python, that you are guaranteed to find on the build machine.
+Moreover, it can be used with various backends, like Ninja, Visual Studio and
+Xcode, which means it can be used to generate a Visual Studio solution that
+you can import there.
+- It forces you to build out-of-source, meaning that no changes (mostly) will
+be done to your source directory, which must contain only the source files of
+your project and not be mixed with other auto-generated files. This also
+allows you to have the project compiled with different options or with
+slightly different code, cleanly separated in different directories.
+
+# Examples of using meson
+
+## Development process
+When I download Rizin, I install it for my user in `~/.local`, so I don't need
+root access to install files. I do this with `meson --prefix=~/.local build;
+ninja -C build install`. After that, I can change the source code however I
+need and then I usually run `ninja` again with `ninja -C build`. Only the
+changed files are re-built. Moreover, `ninja` by default builds files with
+RPATH, which means that the executables/libraries are directly "linked" with
+the dependent libraries that are built. For this reason, most of the times I
```

Comment:
I see what you did there by putting linked in "s but I think it will still be a bit hard to understand what you actually mean. Maybe use something like this:
```
Moreover, running `ninja` by default builds files with explicit
RPATHs, which means that the executables/libraries contain direct references
to the paths of dependent libraries they are linked against so the loader can
then always find them without having to specify `LD_LIBRARY_PATH` or similar.
```